data_diff/diff_tables.py

```diff
@@ -73,11 +96,28 @@ def __post_init__(self):
 
     @property
     def _key_column(self):
-        return ColumnName(self.key_column)
+        return self._quote_column(self.key_column)
 
     @property
     def _update_column(self):
-        return ColumnName(self.update_column)
+        return self._quote_column(self.update_column)
+
+    def _quote_column(self, c):
+        if self.case_sensitive:
+            return self.database.quote(c)
```

Comment:
Instead of disabling quoting (which I think is kinda hard), why not use the case-insensitive dict to get the right name for the column?

Reply:
As long as you always show the quoted form in the errors / exceptions, I think it's quite valid.

Reply:
(1) Like do I think you should use the case-insensitive map to get the "correct" casing and raise an exception if there's ever multiple entries in the map to disable case-insensitivity? Yep.

(2) By "show the quoted form", I mean that in all exceptions that mentions the column, to show `"MY_primARy_KEY"` not `MY_primARy_KEY` or `my_primary_key` 👍🏻

Reply:
Concern is especially when they have different casing between e.g. Postgres and Snowflake. Not sure what happens without this patch for that

Reply:
1) That's already what we're doing..

2) Don't we already do that? 👍🏻

Reply:
Good, wanted to confirm!